src/io/flutter/console/FlutterConsoleHelper.java

```diff
@@ -21,60 +22,106 @@
 import com.intellij.ui.content.ContentManager;
 import com.intellij.ui.content.MessageView;
 import org.jetbrains.annotations.NotNull;
-
-// TODO(devoncarew): These pile up - how to close or recycle an existing one?
-// See: https://github.com/JetBrains/intellij-plugins/blob/8573cd83bb6826e339d60be7b9cdb97add1af2a2/Dart/src/com/jetbrains/lang/dart/ide/actions/DartPubActionBase.java#L203
+import org.jetbrains.annotations.Nullable;
 
 public class FlutterConsoleHelper {
+  private static final Key<FlutterConsoleInfo> FLUTTER_MESSAGES_KEY = Key.create("FLUTTER_MESSAGES_KEY");
 
   /**
    * Attach the flutter console to the process managed by the given processHandler.
    */
-  public static void attach(@NotNull Module module, @NotNull OSProcessHandler processHandler, @NotNull String processTitle) {
-    final ConsoleView console = getConsole(module);
-    console.attachToProcess(processHandler);
-    show(module.getProject(), console, processTitle);
+  public static void attach(@NotNull Project project, @NotNull OSProcessHandler processHandler) {
+    show(project, null, processHandler);
   }
 
   /**
    * Attach the flutter console to the process managed by the given processHandler.
    */
-  public static void attach(@NotNull Project project, @NotNull OSProcessHandler processHandler, @NotNull String processTitle) {
-    final ConsoleView console = getConsole(project);
-    console.attachToProcess(processHandler);
-    show(project, console, processTitle);
+  public static void attach(@NotNull Module module, @NotNull OSProcessHandler processHandler) {
+    show(module.getProject(), module, processHandler);
   }
 
-  private static void show(@NotNull Project project, @NotNull ConsoleView console, String processTitle) {
-    final SimpleToolWindowPanel toolWindowPanel = new SimpleToolWindowPanel(false, true);
-    toolWindowPanel.setContent(console.getComponent());
+  /**
+   * We share one Flutter console per Module, and have one global Flutter console (for tasks which don't
+   * take a module, like 'flutter doctor'). We could revisit to have one global, shared flutter console
+   * instance.
+   *
+   * @param project
+   * @param module
+   * @param processHandler
+   */
+  private static void show(@NotNull Project project, @Nullable Module module, @NotNull OSProcessHandler processHandler) {
+    if (findExistingInfoForCommand(project, module) != null) {
+      final FlutterConsoleInfo info = findExistingInfoForCommand(project, module);
+      info.console.clear();
+      info.console.attachToProcess(processHandler);
+
+      final MessageView messageView = MessageView.SERVICE.getInstance(project);
+      messageView.runWhenInitialized(() -> {
+        final ContentManager contentManager = messageView.getContentManager();
+        contentManager.setSelectedContent(info.content);
+      });
+    }
+    else {
+      ConsoleView console = createConsole(project, module);
```

Comment:
On the last review I decided not to comment on inconsistent use of `final` but it's distracting enough that I think maybe we should consider re-enabling the inspection that enforces it.  Thoughts?